vm/src/obj/objstr.rs

```diff
@@ -1231,18 +1231,22 @@ fn do_cformat_specifier(
             Ok(format_spec.format_number(objint::get_value(&obj)))
         }
         CFormatType::Float(_) => {
-            if objtype::isinstance(&obj, &vm.ctx.float_type()) {
-                Ok(format_spec.format_float(objfloat::get_value(&obj)))
+            let result = if objtype::isinstance(&obj, &vm.ctx.float_type()) {
+                format_spec.format_float(objfloat::get_value(&obj))
             } else if objtype::isinstance(&obj, &vm.ctx.int_type()) {
-                Ok(format_spec.format_float(objint::get_value(&obj).to_f64().unwrap()))
+                format_spec.format_float(objint::get_value(&obj).to_f64().unwrap())
             } else {
                 let required_type_string = "an floating point or integer";
-                Err(vm.new_type_error(format!(
+                return Err(vm.new_type_error(format!(
                     "%{} format: {} is required, not {}",
                     format_spec.format_char,
                     required_type_string,
                     obj.class()
-                )))
+                )));
+            };
+            match result {
```

Comment:
Instead of this `match` statement, you might want to choose the `map_err` function like this:
```rust

if is_float {
    ...
} else if is_int {
    ...
} else {
    ...
}.map_err(|e| vm.new_not_implemented_error(e))
```

Reply:
Do you mean that it should change like you say at Line 1234 to 1246?

Or not, really at Line 1247, I think it doesn't need to change like you say because `format_float` always returns `String` type `Result`.
```rust
pub fn format_float(&self, num: f64) -> Result<String, String>
```

So there is nothing to map like that.

Or, is there any good example for handling `Result` type returns?

Thank you for reviewing :)

Reply:
My comment is an idea to simplify the `match` statement at the end here. Let me try to clarify. You could change this:

```rust
match result {
    Ok(transformed) => Ok(transformed),
    Err(error) => Err(vm.new_not_implemented_error(error)),
}
```

Into this:

```rust
result.map_err(|e| vm.new_not_implemented_error(e))
```

The `map_err` can be used here to transform the type `Result<String, String>` into `PyResult<String>`.

I hope I could explain this idea.

Reply:
I changed that part :)
I hope that's what you explained to me.

Reply:
Looks good to me! Nicely done!